Source/JavaScriptCore/runtime/JSArrayBufferView.h

```diff
@@ -246,16 +354,17 @@ class JSArrayBufferView : public JSNonFinalObject {
 
     VectorPtr m_vector;
     size_t m_length;
-    size_t m_maxByteLength { 0 };
+    size_t m_byteOffset { 0 };
     TypedArrayMode m_mode;
 };
 
-class IdempotentArrayBufferByteLengthGetter;
-
 JSArrayBufferView* validateTypedArray(JSGlobalObject*, JSValue);
-bool isIntegerIndexedObjectOutOfBounds(JSArrayBufferView*, IdempotentArrayBufferByteLengthGetter&);
-std::optional<size_t> integerIndexedObjectLength(JSArrayBufferView*, IdempotentArrayBufferByteLengthGetter&);
-size_t integerIndexedObjectByteLength(JSArrayBufferView*, IdempotentArrayBufferByteLengthGetter&);
+
+inline ArrayBuffer* JSArrayBufferView::existingBufferInButterfly()
+{
+    ASSERT(isWastefulTypedArray(m_mode));
+    return butterfly()->indexingHeader()->arrayBuffer();
+}
```

Comment:
Because of linker failures in GTK / WPE. Some files are not correctly importing JSArrayBufferViewInlines.h, and this cause link failures. Since we do not need to put this to Inlines.h side, it is just placing it to this header to make compilation simpler.